eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -625,24 +814,11 @@ object Graph {
        *
        * @param channels map of all known public channels in the network.
        */
-      def makeGraph(channels: SortedMap[RealShortChannelId, PublicChannel]): DirectedGraph = {
-        // initialize the map with the appropriate size to avoid resizing during the graph initialization
-        val mutableMap = new mutable.HashMap[PublicKey, List[GraphEdge]](initialCapacity = channels.size + 1, mutable.HashMap.defaultLoadFactor)
-
-        // add all the vertices and edges in one go
-        channels.values.foreach { channel =>
-          channel.update_1_opt.foreach(u1 => addToMap(GraphEdge(u1, channel)))
-          channel.update_2_opt.foreach(u2 => addToMap(GraphEdge(u2, channel)))
-        }
-
-        def addToMap(edge: GraphEdge): Unit = {
-          mutableMap.put(edge.desc.b, edge +: mutableMap.getOrElse(edge.desc.b, List.empty[GraphEdge]))
-          if (!mutableMap.contains(edge.desc.a)) {
-            mutableMap += edge.desc.a -> List.empty[GraphEdge]
-          }
-        }
+      def makeGraph(channels: Map[RealShortChannelId, PublicChannel], nodes: Map[PublicKey, NodeAnnouncement]): DirectedGraph = {
+        val edges = channels.values.flatMap(channel =>
+          Seq(channel.update_1_opt.map(GraphEdge(_, channel)), channel.update_2_opt.map(GraphEdge(_, channel))).flatten)
 
-        new DirectedGraph(mutableMap.toMap)
+        DirectedGraph().addVertices(nodes.values).addEdges(edges)
```

Comment:
IIUC, the trade-off is that if we use a map, it makes `addEdge` faster (graph update), but it makes `getIncomingEdgesOf` slower (graph search). When we made that code, people weren't changing their channel update frequently, so it made sense to optimize for the path-finding case. Since node operators now update their channels more frequently, your change could make sense.